jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/kafka/listeners/Asn1DecodedDataRouter.java

```diff
@@ -0,0 +1,194 @@
+package us.dot.its.jpo.ode.kafka.listeners;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.json.JSONObject;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Component;
+import us.dot.its.jpo.ode.coder.OdeBsmDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdeMapDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdePsmDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdeSpatDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdeSrmDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdeSsmDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdeTimDataCreatorHelper;
+import us.dot.its.jpo.ode.context.AppContext;
+import us.dot.its.jpo.ode.kafka.producer.DisabledTopicException;
+import us.dot.its.jpo.ode.kafka.topics.JsonTopics;
+import us.dot.its.jpo.ode.kafka.topics.PojoTopics;
+import us.dot.its.jpo.ode.model.OdeAsn1Data;
+import us.dot.its.jpo.ode.model.OdeBsmData;
+import us.dot.its.jpo.ode.model.OdeLogMetadata;
+import us.dot.its.jpo.ode.model.OdeLogMetadata.RecordType;
+import us.dot.its.jpo.ode.plugin.j2735.J2735DSRCmsgID;
+import us.dot.its.jpo.ode.util.XmlUtils;
+import us.dot.its.jpo.ode.util.XmlUtils.XmlUtilsException;
+
+/**
+ * The Asn1DecodedDataRouter class is a component responsible for processing decoded ASN.1 data
+ * from Kafka topics. It listens to messages on a specified Kafka topic and handles the incoming
+ * data by processing and forwarding it to different topics based on specific criteria.
+ *
+ * <p>This listener is specifically designed to handle decoded data produced by the asn1_codec.
+ * Upon receiving a payload, it uses transforms the payload and then determines the appropriate
```

Comment:
(blocking) Looks like a typo here: 'uses transforms' should probably just be 'transforms'

Reply:
corrected in https://github.com/CDOT-CV/jpo-ode/pull/131/commits/d7f2bbecede1d0da5d3a38ea7db8fce0985373ee